edb/server/tenant.py

```diff
@@ -243,21 +254,35 @@ def set_reloadable_files(
             self._jwt_revocation_list_file = jwt_revocation_list_file
             rv = True
 
+        if isinstance(config_file, str):
+            config_file = pathlib.Path(config_file)
+        if self._config_file != config_file:
+            self._config_file = config_file
+            rv = True
+
         return rv
 
     def set_server(self, server: edbserver.BaseServer) -> None:
         self._server = server
         self.__loop = server.get_loop()
 
     async def load_sidechannel_configs(
-        self, value: Any, *, compiler: edbcompiler.Compiler | None = None
+        self,
+        value: Any,
+        *,
+        compiler: (
+            edbcompiler.Compiler | edbcompiler_pool.AbstractPool | None
+        ) = None,
     ) -> None:
         if compiler is None:
             compiler = self._server.get_compiler_pool()
         result = compiler.compile_structured_config(
-            {"cfg::Config": {"email_providers": value}}, source="magic",
-            allow_nested=True,
+            {"cfg::Config": {"email_providers": value}},
+            "magic",  # source
+            True,  # allow_nested
         )
+        if asyncio.iscoroutine(result):
+            result = await result
```

Comment:
This smells; can we at least document it?

Reply:
Fixed to leverage typing